packages/amplify-graphql-transformer-core/src/transformation/transform.ts

```diff
@@ -308,49 +308,50 @@ export class GraphQLTransform {
   }
 
   public applyOverride = (stackManager: StackManager): AmplifyApiGraphQlResourceStackTemplate => {
-    const stacks: string[] = [];
-    const amplifyApiObj: any = {};
-    stackManager.rootStack.node.findAll().forEach(node => {
-      const resource = node as CfnResource;
-      if (resource.cfnResourceType === 'AWS::CloudFormation::Stack') {
-        stacks.push(node.node.id.split('.')[0]);
-      }
-    });
+    const overrideFilePath = path.join(this.overrideConfig!.overrideDir, 'build', 'override.js');
```

Comment:
we're checking `!_.isEmpty(this.overrideConfig)` in the below line, but I am afraid `this.overrideConfig!.overrideDir` will throw if that's the case here.

Reply:
@phani-srikar thanks for spotting this. It was working e2e because callers have redundant check.
I refactored those conditionals a bit to simplify the flow. I.e. now non-empty check is bundled with overrideFlag check through `?` operator.
I also inverted logic to shortcircuit this method.